src/scripts/dom.ts

```diff
@@ -77,156 +83,132 @@ function getDayOfWeek(num) {
   return dayNames[num];
 }
 
-function getWeatherDescription(weather) {
+function getWeatherDescription(weather: string): string {
   if (!weather) return "";
 
   switch (weather) {
     case "clear":
       return "Clear";
-      break;
     case "cloudy":
       return "Overcast";
-      break;
     case "lightrain":
       return "Light rain";
-      break;
     case "pcloudy":
       return "Partly cloudy";
-      break;
     case "ts":
       return "Isolated thunderstorms";
-      break;
     case "snow":
       return "Snow";
-      break;
     case "rain":
       return "Rain";
-      break;
     case "tsrain":
       return "Thunderstorm";
-      break;
     case "mcloudy":
       return "Cloudy";
-      break;
     case "humid":
       return "Foggy";
-      break;
     case "oshower":
       return "Occasional showers";
-      break;
     case "ishower":
       return "Isolated showers";
-      break;
     case "lightsnow":
       return "Light snow";
-      break;
     case "rainsnow":
       return "Ice pellets/Freezing rain";
-      break;
     default:
       return "Windy";
   }
 }
 
-function getTempRangesString(mintemp, maxtemp) {
+function getTempRangesString(mintemp: number, maxtemp: number): string {
   return `${mintemp}\xB0C ~ ${maxtemp}\xB0C`;
 }
 
-function set7dayForecast(forecast_data) {
-  let seven_day_forecast = document.getElementById("seven_day_forecast");
-  seven_day_forecast.innerHTML = "";
-  forecast_data.forEach((day) => {
-    let forecast_element = setDay(day);
-    seven_day_forecast.appendChild(forecast_element);
-  });
+function set7dayForecast(forecast_data: Series[]): void {
+  let seven_day_forecast: HTMLElement | null =
+    document.getElementById("seven_day_forecast");
+  if (seven_day_forecast) {
```

Comment:
Instead of setting the `seven-day-forecast-element` to be either of type `HTMLElement | null` I would adives you to do this:
```ts
const seven_day_forecast = document.getElementById("seven-day-forecast");
// This is a guard clause
if (!seven_day_forecast) {
  return;
}

// Then do the rest of your code for when seven_day_forecast exists (is not null).
  
```